src/duckling/game/drawable/drawable-drawer.ts

```diff
@@ -23,60 +33,52 @@ import {Rectangle} from './rectangle';
  * @param  entity The entity with the drawable attribute
  * @return DisplayObject that contains the drawn DrawableAttribute
  */
-export function drawDrawableAttribute(entity : Entity, assetService : AssetService) : DisplayObject {
-    var positionAttribute = getPosition(entity);
-    var drawableAttribute = getDrawableAttribute(entity);
+export function drawDrawableAttribute(entity : Entity, assetService : AssetService) : DrawnConstruct {
+    let positionAttribute = getPosition(entity);
+    let drawableAttribute = getDrawableAttribute(entity);
     if (!positionAttribute || !drawableAttribute.topDrawable) {
         return null;
     }
 
-    var drawable = drawDrawable(drawableAttribute.topDrawable, assetService);
-    if (!drawable) {
-        return null;
-    }
+    let drawable = drawDrawable(drawableAttribute.topDrawable, assetService);
+    setNonInteractive(drawable);
+    setPosition(drawable, positionAttribute.position);
 
-    drawable.position.x = positionAttribute.position.x;
-    drawable.position.y = positionAttribute.position.y;
     return drawable;
 }
 
-function drawDrawable(drawable : Drawable, assetService : AssetService) : DisplayObject {
+function drawDrawable(drawable : Drawable, assetService : AssetService) : DrawnConstruct {
     if (drawable.inactive) {
-        return null;
+        return new DisplayObject();
     }
 
-    var drawableContainer = new Container();
-    switch (drawable.type) {
+    let drawnObject : DrawnConstruct = new DisplayObject();
```

Comment:
You are allocating a DisplayObject every time even if you don't need it. You should only allocate in the default case of the switch.